scripts/upgrade_admin_server.sh

```diff
@@ -65,8 +65,20 @@ upgrade_admin_server()
     if [ $ret != 0 ]; then
         # In the failed case, crowbar should tell user to check zypper logs,
         # fix the errors and continue admin server manually
-        echo "zypper dist-upgrade has failed with $ret, check zypper logs"
+        local errmsg="zypper dist-upgrade has failed with $ret, check zypper logs"
+        echo $errmsg
         echo "$ret" > $UPGRADEDIR/admin-server-upgrade-failed
+        # The status of the upgrade needs to be set in case of an error
+        # so we can just use the upgrade_status library to set the status to "failed"
+        # otherwise the status of the admin_upgrade would stay at "running" which would prevent
+        # continuation of the upgrade even if the admin upgrade got fixed manually
+        ruby -e "
```

Comment:
AFAIK, we do not need it for any other case than this. I assume @skazi0 is suggesting it to make things clearer